crypto/keys/internal/ecdsa/privkey_internal_test.go

```diff
@@ -60,7 +63,39 @@ func (suite *SKSuite) TestSign() {
 		require.False(suite.pk.VerifySignature(msg, sigCpy))
 	}
 
+	// mutate the signature by scalar neg'ing the s value
+	// to give a high-s signature, valid ECDSA but should
+	// be invalid with Cosmos signatures.
+	// code mostly copied from privkey/pubkey.go
+
+	// extract the r, s values from sig
+	r := new(big.Int).SetBytes(sig[:32])
+	low_s := new(big.Int).SetBytes(sig[32:64])
+
+	// flip the s value into high order of curve P256
+	// leave r untouched!
+	high_s := new(big.Int).Mod(new(big.Int).Neg(low_s), elliptic.P256().Params().N)
+
+	// turn these big ints back into 0-padded bytes
+	// replicating what happens in signature
+	rBytes := r.Bytes()
+	sBytes := high_s.Bytes()
+	sigBytes := make([]byte, 64)
+
+	// 0 pad the byte arrays from the left if they aren't big enough.
+	copy(sigBytes[32-len(rBytes):32], rBytes)
+	copy(sigBytes[64-len(sBytes):64], sBytes)
```

Comment:
could you reuse the functions from `privkey.go`? This will increase code coverage - which is what we need:
```
sigBytes := signatureRaw(r, high_s)
```

Reply:
I didn't reuse them because:

1. they are private, and at least my use of unit tests has always been only to test the external interface. Some engineers believe that either methods should be made public, or they should not be unit tested from the outside. Dunno if cosmos takes an opinion on that?
2. they don't take the same inputs that I need in this different context.
3. In general, the tests I did here are kind of hackish, and I'd prefer to see static test vectors for signatures, beyond the basic sign/verify tests, to allow for better reproducibility.

Reply:
Re 1. This file is meant to test private (internal) functions as well. You don't need to export them. It's good that they are private.

Re 2. Which input doesn't work here?

Re 3. If we copy or redo the code then we may miss some potential error which could be found by the test.

Reply:
I totally spaced on that it _already_ had private access, and only saw it after you explained on the call :) So I just called the private function.